dask/dataframe/io/parquet.py

```diff
@@ -0,0 +1,170 @@
+import os
+import struct
+
+from toolz import first
+
+import dask
+from dask import delayed, compute
+from dask.bytes.core import open_files, OpenFileCreator
+import dask.dataframe as dd
+from dask import delayed
+
+try:
+    import fastparquet
+    from fastparquet import parquet_thrift
+    default_encoding = parquet_thrift.Encoding.PLAIN
+except:
+    fastparquet = False
+    default_encoding = None
+
+
+def read_parquet(path, columns=None, filters=[], categories=None, index=None,
+        **kwargs):
+    """ Read Dask DataFrame from ParquetFile
+
+    This reads a directory of Parquet data into a Dask.dataframe, one file per
+    partition.  It selects the index among the sorted columns if any exist.
+
+    Parameters
+    ----------
+    path : string
+        Source directory for data.
+        Prepend with protocol like ``s3://`` or ``hdfs://`` for remote data.
+    columns: list or None
+        List of column names to load
+    filters: list
+        List of filters to apply, like ``[('x', '>' 0), ...]``
+    index: string or None
+        Name of index column to use if that column is sorted
+
+    Examples
+    --------
+    >>> df = read_parquet('s3://bucket/my-parquet-data')  # doctest: +SKIP
+
+    See Also
+    --------
+    to_parquet
+    """
+    if fastparquet is False:
+        raise ImportError("fastparquet not installed")
+    myopen = OpenFileCreator(path, compression=None, text=False)
+
+    try:
+        pf = fastparquet.ParquetFile(path + '/_metadata',
+                                     open_with=myopen,
+                                     sep=myopen.fs.sep)
+        root = path
+    except:
+        pf = fastparquet.ParquetFile(path, open_with=myopen, sep=myopen.fs.sep)
+        root = os.path.dirname(path)  # TODO: this might fail on S3 + Windows
```

Comment:
Please feel free to push to this branch directly.  As a maintainer you should have write privileges.